helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkConnectionConfig.java

```diff
@@ -0,0 +1,190 @@
+    RealmAwareZkClient.RealmAwareZkConnectionConfig zkConnectionConfig =
+            new RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder()
+                    .setRoutingDataSourceEndpoint(_msdsEndpoint + "," + ZK_PREFIX + ZK_START_PORT)
+                    .setRoutingDataSourceType(RoutingDataReaderType.HTTP_ZK_FALLBACK.name()).build();
+    HelixAdmin helixAdminBuilder =
+            new ZKHelixAdmin.Builder().setRealmAwareZkConnectionConfig(zkConnectionConfig).build();
+    _zkHelixAdmin =
+            new ZKHelixAdmin.Builder().setRealmAwareZkConnectionConfig(zkConnectionConfig).build();
+
+    String participantNamePrefix = "Node_";
+    int numParticipants = 5;
+    //createParticipantsAndVerify(helixAdminZkAddr, numParticipants, participantNamePrefix);
+    createParticipantsAndVerify(helixAdminBuilder, numParticipants, participantNamePrefix);
+
+    // Create mock controller and participants for next tests
+    for (String cluster : CLUSTER_LIST) {
+      // Start a controller
+      // Note: in multiZK mode, ZK Addr is ignored
+      RealmAwareZkClient.RealmAwareZkConnectionConfig zkConnectionConfigCls =
+              new RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder()
+                      .setRoutingDataSourceEndpoint(_msdsEndpoint + "," + ZK_PREFIX + ZK_START_PORT)
+                      .setRoutingDataSourceType(RoutingDataReaderType.HTTP_ZK_FALLBACK.name())
+                      .setZkRealmShardingKey("/" + cluster).build();
+      HelixManagerProperty.Builder propertyBuilder = new HelixManagerProperty.Builder();
+      HelixManagerProperty helixManagerProperty =
+              propertyBuilder.setRealmAWareZkConnectionConfig(zkConnectionConfigCls).build();
+      ClusterControllerManager mockController =
+              new ClusterControllerManager(cluster, helixManagerProperty);
+      mockController.syncStart();
+      MOCK_CONTROLLERS.put(cluster, mockController);
+
+      for (int i = 0; i < numParticipants; i++) {
+        // Note: in multiZK mode, ZK Addr is ignored
+        InstanceConfig instanceConfig = new InstanceConfig(participantNamePrefix + i);
+        helixAdminBuilder.addInstance(cluster, instanceConfig);
+        MockParticipantManager mockNode =
+                new MockParticipantManager(cluster, participantNamePrefix + i, helixManagerProperty, 10,
+                        null);
+
+        // Register task state model for task framework testing in later methods
+        Map<String, TaskFactory> taskFactoryReg = new HashMap<>();
+        taskFactoryReg.put(MockTask.TASK_COMMAND, MockTask::new);
+        // Register a Task state model factory.
+        StateMachineEngine stateMachine = mockNode.getStateMachineEngine();
+        stateMachine
+                .registerStateModelFactory("Task", new TaskStateModelFactory(mockNode, taskFactoryReg));
+
+        mockNode.syncStart();
+        MOCK_PARTICIPANTS.add(mockNode);
```

Comment:
If we have `MOCK_PARTICIPANTS` as a non-static var, we probably don't need to repeat this test in 2 classes?